pkg/query/querier.go

```diff
@@ -365,6 +376,52 @@ func (q *querier) Select(ctx context.Context, _ bool, hints *storage.SelectHints
 }
 
 func (q *querier) selectFn(ctx context.Context, hints *storage.SelectHints, ms ...*labels.Matcher) (storage.SeriesSet, storepb.SeriesStatsCounter, error) {
+	if q.enablePreAggregationLabelRewrite {
+		hasNameLabel, needLabelRewrite := false, false
+		var nameLabelMatcher *labels.Matcher
+		for _, m := range ms {
+			if m.Name == labels.MetricName {
+				hasNameLabel = true
+				if m.Type == labels.MatchEqual {
+					for _, suffix := range q.aggregationStandardSuffixes {
+						if strings.HasSuffix(m.Value, suffix) {
+							needLabelRewrite = true
+							nameLabelMatcher = m
+							break
+						}
+					}
+					break
+				} else {
+					level.Info(q.logger).Log("msg", "Skipping aggregation label rewrite attempt", "reason", "non-equal name matcher", "name_matcher", m)
```

Comment:
the log will be very verbose as query QPS is high, please only consider debug level